File: demo-apps/TradecoreDemoObjectiveC/TradecoreDemoObjectiveC/BaseVC/InstreamBaseVC/InstreamBaseViewController.m
//
//  InstreamBaseViewController.m
//  TradecoreDemoObjectiveC
//
//  Created by Olena Stepaniuk on 15.05.2025.
//

#import "InstreamBaseViewController.h"

@interface InstreamBaseViewController ()
@end

@implementation InstreamBaseViewController

- (NSString *)nibName {
    return @"InstreamBaseViewController";
}

- (instancetype)initWithAdSize:(CGSize)adSize {
    self = [super initWithNibName:self.nibName bundle:nil];
    if (self) {
        _adSize = adSize;
    }
    return self;
}

- (instancetype)init {
    return [self initWithAdSize:CGSizeMake(300, 250)];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.playButton.layer.zPosition = CGFLOAT_MAX;
    [self.reloadButton setEnabled:false];
}

- (IBAction)onReloadTapped:(id)sender {
    [self.reloadButton setEnabled:false];
}

- (IBAction)onPlayTapped:(id)sender {
    [self.playButton setHidden:true];
}

@end

File: demo-apps/TradecoreDemoAppSwift/TradecoreDemoAppSwift/BaseVC/InstreamBaseVC/InstreamBaseViewController.swift
//
//  InstreamBaseViewController.swift
//  TradecoreDemoAppSwift
//
//  Created by Olena Stepaniuk on 14.05.2025.
//

import UIKit

fileprivate let instreamBaseViewControllerNibName = "InstreamBaseViewController"

class InstreamBaseViewController: UIViewController {
    
    @IBOutlet weak var containerView: UIView!
    
    @IBOutlet weak var containerWidth: NSLayoutConstraint!
    @IBOutlet weak var containerHeight: NSLayoutConstraint!
    
    @IBOutlet weak var playButton: UIButton!
    @IBOutlet weak var reloadButton: UIButton!
    
    private(set) var adSize = CGSize(width: 300, height: 250)
    
    convenience init(adSize: CGSize) {
        self.init(nibName: instreamBaseViewControllerNibName, bundle: nil)
        self.adSize = adSize
    }
    
    convenience init() {
        self.init(nibName: instreamBaseViewControllerNibName, bundle: nil)
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        playButton.layer.zPosition = CGFloat.greatestFiniteMagnitude
        reloadButton.isEnabled = false
    }
    
    @IBAction func onReloadTapped(_ sender: Any) {
        reloadButton.isEnabled = false
    }
    
    @IBAction func onPlayTapped(_ sender: Any) {
        playButton.isHidden = true
    }
}
